features/step_definitions/file_steps.coffee

```diff
@@ -4,18 +4,22 @@ path = require 'path'
 
 module.exports = ->
 
-  @Given /^I have no test files$/, (done) ->
-    done() # No op for readability
+  @Given /^my project has no test files$/, (done) ->
+    done()
 
 
-  @Given /^I have 1 (passing|failing) test$/, (status, done) ->
-    outputTestFile path.join(@tmpDir, 'example_spec.coffee'), passing: status is 'passing', done
+  @Given /^my project has a (passing|failing) test$/, (status, done) ->
+    outputTestFile path.join(@tmpDir, 'example_spec.coffee'),
+                   passing: status is 'passing',
+                   done
```

Comment:
Very cool, I like where you are going with this! I agree that temp variables can add documentation through their names in certain cases. While implementing your suggestion, I noticed the following code smells, however:

* parsing Gherkin is responsibility of the Cucumber step definition or its helpers, not the tool that creates passing or failing test files. 
* in the step definition, we now first introduce a temp variable to increase readability, and then later have to use CoffeeScript tricks to get rid of the redundancy between our temp variable name and a named parameter with the same name. Ultimately we already have named parameters for documentation here, so the temp variables aren't really adding anything. 
* the step definition still contains a pretty complex line that is hard to read. In this particular situation, a different formatting reads better.

So I ended up with this implementation:

```coffeescript
# file_steps.coffee
isPassing = (status) -> status is 'passing'

module.exports = ->

  @Given /^my project has a (passing|failing) test$/, (status, done) ->
    createTest
      filePath: path.join(@tmpDir, 'example_spec.coffee')
      passing: isPassing(status)
      done

# file_helpes.coffee
createTest = ({filePath, passing}, done) ->
  fsExtra.outputFile filePath, testContent(passing), done
```

This seems even cleaner, more redundancy free, and the responsibilities better allocated. What do you think?